Refactoring in parser tests

khieta opened the pull request:
## Description of changes

I was taking a look at `ParseErrors` as part of my assigned work for #745, and realized that I wanted to make some improvements before officially exposing the type. I'm adding one of the uninteresting improvements now to make the diff of my future PR(s) smaller. 
* Refactored the CST->AST tests to improve code re-use. This will make the diff for my future PR (which needs to update tests) much smaller.

## Issue #, if available

## Checklist for requesting a review

The change in this PR is (choose one, and delete the other options):

- [X] A breaking change requiring a major version bump to `cedar-policy` (e.g., changes to the signature of an existing API).

(I think it's breaking due to the `RestrictedExprParseError` -> `ParseErrors` change)

I confirm that this PR (choose one, and delete the other options):

- [X] Does not update the CHANGELOG because my change does not significantly impact released code.

I confirm that [`cedar-spec`](https://github.com/cedar-policy/cedar-spec) (choose one, and delete the other options):

- [X] Does not require updates because my change does not impact the Cedar formal model or DRT infrastructure.

shaobo-he-aws reviewed (approved):
LGTM

cdisselkoen commented on cedar-policy-core/src/parser/cst_to_ast.rs:
```diff
@@ -2536,380 +2536,404 @@ mod tests {
-        });
+        );
+        // annotation duplication (anno)
+        assert_eq!(errs.len(), 1, "{:?}", miette::Report::new(errs));
```
maybe worth improving this check to an `ExpectedErrorMessage`? (Doesn't have to be in this PR)

also for some other cases below

cdisselkoen reviewed (approved):
Looks great.  Good work was done here

khieta commented:
Thanks for the comments everyone! In the latest commit I reverted the `RestrictedExprParseError` change -- I want to play around with @john-h-kastner-aws's suggestion, but I'll do this in a future PR. I also added several `ExpectedErrorMessage`s, per @cdisselkoen's suggestion.

john-h-kastner-aws reviewed (approved):
awesome
